src/Server/Utils/Strategies/RegexResponseStrategy.php

```diff
@@ -67,13 +75,25 @@ private function bodyConditionIsRegex($expectation)
     private function fillWithUrlMatches($expectation, $httpRequest, $responseBody)
     {
         if ($this->urlConditionIsRegex($expectation)) {
-            $responseBody = preg_replace('/\$\{url\.(\d+)\}/', '\$$1', $responseBody);
-
-            return preg_replace(
-                $expectation->getRequest()->getUrl()->getValue(),
-                $responseBody,
-                $httpRequest->getUri()->__toString()
+            $pattern = preg_replace(
+                '/^(.)\^/',
+                '$1',
+                $expectation->getRequest()->getUrl()->getValue()
             );
+            preg_match(
+                $pattern,
+                $httpRequest->getUri()->__toString(),
+                $matches
+            );
+
+            if (isset($matches[1])) {
```

Comment:
What do you think about extracting this block of code to a private method and parametrize the left part of the replacement? this way we can reuse it for the body replacement and don't have duplicated code in both methods.

Reply:
Yes, it's good idea.